custom_components/midea_ac_lan/climate.py

```diff
@@ -363,107 +380,119 @@ def set_swing_mode(self, swing_mode: str) -> None:
+            else C3Attributes.zone2_power
+        )
 
     @property
-    def supported_features(self):
+    def supported_features(self) -> ClimateEntityFeature:
         features = ClimateEntityFeature.TARGET_TEMPERATURE
         if (MAJOR_VERSION, MINOR_VERSION) >= (2024, 2):
             features |= ClimateEntityFeature.TURN_OFF | ClimateEntityFeature.TURN_ON
         return features
 
     @property
-    def target_temperature_step(self):
+    def target_temperature_step(self) -> float:
         return (
             PRECISION_WHOLE
             if self._device.get_attribute(C3Attributes.zone_temp_type)[self._zone]
             else PRECISION_HALVES
         )
 
     @property
-    def min_temp(self):
-        return self._device.get_attribute(C3Attributes.temperature_min)[self._zone]
+    def min_temp(self) -> float:
+        return cast(
+            float, self._device.get_attribute(C3Attributes.temperature_min)[self._zone]
+        )
 
     @property
-    def max_temp(self):
-        return self._device.get_attribute(C3Attributes.temperature_max)[self._zone]
+    def max_temp(self) -> float:
+        return cast(
+            float, self._device.get_attribute(C3Attributes.temperature_max)[self._zone]
+        )
 
     @property
-    def target_temperature_low(self):
-        return self._device.get_attribute(C3Attributes.temperature_min)[self._zone]
+    def target_temperature_low(self) -> float:
+        return cast(
+            float, self._device.get_attribute(C3Attributes.temperature_min)[self._zone]
+        )
 
     @property
-    def target_temperature_high(self):
-        return self._device.get_attribute(C3Attributes.temperature_max)[self._zone]
+    def target_temperature_high(self) -> float:
+        return cast(
+            float, self._device.get_attribute(C3Attributes.temperature_max)[self._zone]
+        )
 
-    def turn_on(self):
+    def turn_on(self, **kwargs: Any) -> None:
         self._device.set_attribute(attr=self._power_attr, value=True)
 
-    def turn_off(self):
+    def turn_off(self, **kwargs: Any) -> None:
         self._device.set_attribute(attr=self._power_attr, value=False)
 
     @property
     def hvac_mode(self) -> HVACMode:
         if self._device.get_attribute(self._power_attr):
-            return self._modes[self._device.get_attribute(C3Attributes.mode)]
+            return cast(
+                HVACMode, self._modes[self._device.get_attribute(C3Attributes.mode)]
+            )
         else:
             return HVACMode.OFF
 
     @property
-    def target_temperature(self):
-        return self._device.get_attribute(C3Attributes.target_temperature)[self._zone]
+    def target_temperature(self) -> float:
+        return cast(
+            float,
+            self._device.get_attribute(C3Attributes.target_temperature)[self._zone],
+        )
 
     @property
-    def current_temperature(self):
-        return None
+    def current_temperature(self) -> float:
+        return 0
```

Comment:
Returning 0 is incorrect, return type should be `-> float | None`

Reply:
If you want None, then we need to change the superclass as well ;-)

Reply:
In superclass, it's `float | None`

https://github.com/home-assistant/core/blob/ca89d22a34884b789a4154a3e426f93243041042/homeassistant/components/climate/__init__.py#L603